Initiate integration module snakefile

cbethell opened the pull request:
**Issue Addressed**
Closes #341

**What is the purpose of these changes?**
<!--Provide some background on the changes proposed.-->
This PR initiates the data integration snakemake workflow by adding `integration.snakefile`.

**What changes did you make?**
<!--Describe the concrete changes that you made or additional features that were added, be as detailed as possible in the steps you took.-->
In this PR,

- The `integration.snakefile` is added with the `merge-sce` rule that runs the first part of the workflow (the `merge-sce.R` script) 
- A `integration_config.yaml` file is added with the variables that need to be supplied to the snakefile
- The `envs/scpca-renv.yaml` file is updated to include the required `batchelor` bioconductor package.

**Any comments, concerns, or questions important for reviewers**
- Does the PR adequately address #341?
- More importantly, can you run the code in this PR? 
There is a new bioconductor package required here, `batchelor` -- after updating `scpca-renv.yaml` and `renv.lock` with this package, I encounter the following error when trying to run `integration.snakefile` and/or `setup_envs.sh`:

```
Encountered problems while solving:
  - nothing provides requested bioconductor-batchelor 1.14.1**
```

Am I maybe missing a step here for adding a new package? cc: @allyhawkins @jashapiro 

**Checklist**

Place an `x` in all boxes that you have completed.

- [x] I have run the most recent version of the code
- [ ] If I am adding in reports or generating any plots, I have reviewed the necessary reports
- [ ] I have added all necessary documentation (if applicable)

jashapiro commented on envs/scpca-renv.yaml:
```diff
@@ -5,6 +5,7 @@ channels:
   - defaults
 dependencies:
   - bioconductor-annotationdbi=1.60.0
+  - bioconductor-batchelor=1.14.1
```
It looks like the latest version available on bioconda is 1.14.0
```suggestion
  - bioconductor-batchelor=1.14.0
```
https://bioconda.github.io/recipes/bioconductor-batchelor/README.html?highlight=batchelor#package-package%20&#x27;bioconductor-batchelor&#x27;

cbethell replied:
Ahh! Yes that did the trick, thank you 🎉

allyhawkins commented on config/integration_config.yaml:
```diff
@@ -0,0 +1,5 @@
+# All parameters included in this file can be altered at the command line using the `--config` flag or by editing this file directly.
+
+### Optional integration module specific parameters
+project_metadata: "example-data/project-metadata/example-integration-library-metadata.tsv"
```
Hmm I think we might want to actually change this to be `integration_project_metadata`? That way it's distinct from the metadata file that is for the other workflows and we can still use our main config file.

allyhawkins commented on integration.snakefile:
```diff
@@ -0,0 +1,37 @@
+import pandas as pd
+
+configfile: "config/integration_config.yaml"
```
If you take my earlier suggestion, then you can use both configs here and grab the results directory from that. 
```suggestion
configfile: "config/config.yaml"
configfile: "config/integration_config.yaml"
```

allyhawkins commented on config/integration_config.yaml:
```diff
@@ -0,0 +1,5 @@
+
+### Optional integration module specific parameters
+project_metadata: "example-data/project-metadata/example-integration-library-metadata.tsv"
+results_dir: "example-results"
```
```suggestion
```
To be consistent with the other workflows, we should use the results directory we define in the main config file and keep the module config file specific to any parameters unique to integration.

allyhawkins commented on integration.snakefile:
```diff
@@ -0,0 +1,37 @@
+  samples_information = pd.read_csv(config['project_metadata'], sep='\t', index_col=False)
+
+  # get a list of the file paths and integration groups
+  FILEPATHS = list(samples_information['processed_sce_filepath'])
```
```suggestion
```
I don't think you need this actually, because we don't reference it anywhere in the snakemake workflow. The only time the file paths are used is in the Rscript.

allyhawkins reviewed (commented):
This mostly looks good, I just had a few suggestions for cleaning up the config file and keeping things consistent with other modules.

cbethell commented:
Thanks @allyhawkins for the refactoring suggestions, I've implemented them and also added a couple parameters to the `integration_config.yaml` file that I missed the first time around. I have successfully run things with these new updates so this is ready for another look!

allyhawkins reviewed (approved):
Looks good 👍 just one minor comment
